rasa/core/brokers/kafka.py

```diff
@@ -20,6 +25,7 @@ def __init__(
         partition_by_sender: bool = False,
         sasl_username: Optional[Text] = None,
         sasl_password: Optional[Text] = None,
+        sasl_mechanism: Optional[Text] = "PLAIN",
```

Comment:
Optional: I would actually make it `enum.Enum` instead of `Text`, since here you actually provide the set of allowed SASL mechanisms. 

Something like that:

```
class SASLMechanism(enum.Enum):
    PLAIN = 0
    GSSAPI = 1
    ...
    
    def __str__(self) -> Text:
        """Returns a string representation of the object."""
        return self.name.lower()